packages/web-components/src/components/ic-top-navigation/ic-top-navigation.tsx

```diff
@@ -312,17 +320,21 @@ export class TopNavigation {
             <header role="banner">
               <div class="top-panel-container">
                 <div class="app-details-container">
-                  {hasTitle && (
-                    <a class="title-link" href={this.href}>
+                  {(hasTitle || isSlotUsed(this.el, "app-title")) && (
+                    <Component class="title-link" {...attrs}>
                       {this.hasAppIcon && (
                         <div class="app-icon-container" aria-hidden="true">
                           <slot name="app-icon" />
                         </div>
                       )}
                       <ic-typography variant={appTitleVariant}>
-                        <h1 class="title-wrap">{this.appTitle}</h1>
+                        {isSlotUsed(this.el, "app-title") ? (
+                          <slot name="app-title"></slot>
+                        ) : (
+                          <h1 class="title-wrap">{this.appTitle}</h1>
+                        )}
```

Comment:
I think it would be worth moving the `<h1>` out of this block so it's always rendered - at the moment, if a slot is used, there would be no `<h1>` on the page so it might be confusing for screen reader users understanding the structure. This would also ensure the 'title-wrap' class is applied to the slot as well so it has the same wrapping / hyphenation as when using the app-title prop

```suggestion
                      <h1 class="title-wrap">  
                        {isSlotUsed(this.el, "app-title") ? (
                          <slot name="app-title"></slot>
                        ) : (
                          {this.appTitle}
                        )}
                      </h1>
```